src/bp/core/services/cms.ts

```diff
@@ -193,25 +193,22 @@ export class CMSService implements IDisposeOnExit {
     const dbElements = await query.offset(from)
     const elements: ContentElement[] = dbElements.map(this.transformDbItemToApi)
 
-    if (language) {
-      return elements.map(el => {
-        return { ...el, formData: this.getOriginalProps(el.formData, this.getContentType(el.contentType), language) }
-      })
-    }
-    return elements
+    return Promise.map(elements, el => this.translateIfRequired(el, language))
   }
 
-  async getContentElement(botId: string, id: string): Promise<ContentElement> {
+  async getContentElement(botId: string, id: string, language?: string): Promise<ContentElement> {
     const element = await this.memDb(this.contentTable)
       .where({ botId, id })
       .get(0)
 
-    return this.transformDbItemToApi(element)
+    return this.translateIfRequired(this.transformDbItemToApi(element), language)
```

Comment:
i'd rather like you to check if the language param is defined here instead of in the function. thus making this thing more explicit.
```js
const serializedEl = this.transformDbItemToApi(element)
return language ? this.translateElem(serializedEl) : serializedEl
```

Reply:
It was like this at first but i wanted to keep it simple, but you're right. i'll change that